cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/PushOperation.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.sourcecontrol;
+
+import io.cdap.cdap.api.service.operation.LongRunningOperation;
+import io.cdap.cdap.proto.operationrun.OperationError;
+import io.cdap.cdap.proto.operationrun.OperationMeta;
+import io.cdap.cdap.sourcecontrol.operationrunner.PushAppOperationRequest;
+import java.lang.reflect.Type;
+import java.util.function.Consumer;
+
+/**
+ * {@link LongRunningOperation} for scm push.
+ */
+public class PushOperation implements LongRunningOperation<PushAppOperationRequest> {
+
+  @Override
+  public Type getRequestType() {
+    return null;
```

Comment:
Is it a TODO?

Reply:
Yes the PushOperation will be implemented in a separate PR. This is added to setup and test the mapbinder. I have added a TODO top of the class declaration